webpack/assets/javascripts/react_app/components/Editor/Editor.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+// import PropTypes from 'prop-types';
+
+import DiffContainer from '../DiffView/DiffContainer';
+import EditorView from './components/EditorView';
+import EditorNavbar from './components/EditorNavbar';
+import EditorModal from './components/EditorModal';
+import './editor.scss';
+
+class Editor extends React.Component {
+  componentDidMount() {
+    const {
+      data: { template, locked, type },
+      changeState,
+      readOnly,
+      isMasked,
+      selectedView,
+      isRendering,
+    } = this.props;
+
+    changeState('value', template || '');
+    if (readOnly !== locked) {
+      if (locked === true) changeState('readOnly', true);
+      else changeState('readOnly', false);
+    }
+    if (isMasked && type === 'templates') changeState('isMasked', false);
+    if (selectedView !== 'input') changeState('selectedView', 'input');
+    if (isRendering) changeState('isRendering', false);
```

Comment:
wouldn't it be better to use some sort of initialization action?

Reply:
Great suggestion